```
Compile [ 97.9%]: i2c_api.c
[Error] i2c_api.c@432,16: no member named 'sda_func' in 'struct i2c_s'
[Error] i2c_api.c@433,16: no member named 'scl_func' in 'struct i2c_s'
[Error] i2c_api.c@507,37: no member named 'sda_func' in 'struct i2c_s'
[Error] i2c_api.c@508,37: no member named 'scl_func' in 'struct i2c_s'
[ERROR] ./mbed-os/targets/TARGET_STM/i2c_api.c:432:16: error: no member named 'sda_func' in 'struct i2c_s'
        obj_s->sda_func = pinmap->sda_function;
        ~~~~~  ^
./mbed-os/targets/TARGET_STM/i2c_api.c:433:16: error: no member named 'scl_func' in 'struct i2c_s'
        obj_s->scl_func = pinmap->scl_function;
        ~~~~~  ^
./mbed-os/targets/TARGET_STM/i2c_api.c:507:37: error: no member named 'sda_func' in 'struct i2c_s'
    pin_function(obj_s->sda, obj_s->sda_func);
                             ~~~~~  ^
./mbed-os/targets/TARGET_STM/i2c_api.c:508:37: error: no member named 'scl_func' in 'struct i2c_s'
    pin_function(obj_s->scl, obj_s->scl_func);
                             ~~~~~  ^
4 errors generated.
```

This is the errors there for DISCO_L475VG_IOT01A board.

--- a/targets/TARGET_STM/stm_i2c_api.h
+++ b/targets/TARGET_STM/stm_i2c_api.h
@@ -82,6 +82,11 @@ struct i2c_s {
     /// Specifies which events (the I2C_EVENT_xxx defines) can be passed up to the application from the IRQ handler
     uint8_t available_events;
 #endif
+
+#if STATIC_PINMAP_READY
+    int sda_func;
+    int scl_func;
+#endif
 };
 
 #endif

